src/libsyntax/feature_gate.rs

```diff
@@ -638,6 +638,12 @@ pub const KNOWN_ATTRIBUTES: &'static [(&'static str, AttributeType, AttributeGat
                                          is an experimental feature",
                                         cfg_fn!(proc_macro))),
 
+    ("proc_macro_attributes", Normal, Gated(Stability::Unstable,
+                                            "proc_macro",
+                                            "the `#[proc_macro_attributes]` attribute \
+                                             is an experimental feature",
+                                            cfg_fn!(proc_macro))),
```

Comment:
We don't need to add a new kind of attribute -- extending `proc_macro_derive` is fine.